infra/nightly-resources/index.js

```diff
@@ -1,46 +1,194 @@
+            }
+            
+            // the branch is now either the latest main run, or if on main the previous main run
+            // return it
+            return i;
+        }
     }
-    groupedBy[obj.benchmark].push(obj);
-  });
-  let benchmarkNames = Object.keys(groupedBy);
-
-  // for each benchmark, add a table with a name, and a subt-able of each "run method" execution
-  benchmarkNames
-      .forEach((benchName) => {
-        let benchmark = groupedBy[benchName]
-        let toParse = {name: benchName, "Executions ": {data: []}};
-        benchmark.forEach((b) => {
-          let results = b.hyperfine.results[0];
-          toParse["Executions "].data.push({
-            name: b.runMethod,
-            "# Instructions": b.total_dyn_inst,
-            min: tryRound(results.min),
-            max: tryRound(results.max),
-            mean: tryRound(results.mean),
-            median: tryRound(results.median),
-            stddev: tryRound(results.stddev),
-          });
-        });
-
-        parsed.push(toParse);
-      });
-
-  parsed.sort((l, r) => {
-      if (l.name < r.name) { return -1; }
-      if (l.name > r.name) { return 1; }
-      return 0;
-  });
-  container.innerHTML = ConvertJsonToTable(parsed);
+    throw new Error("Couldn't find a benchmark run from main for comparison");
+}
+
+async function getBench(bench) {
+    const resp = await fetch(bench.url + "data/profile.json");
+    const benchData = await resp.json();
+    return groupByBenchmark(benchData)
+}
+
+function groupByBenchmark(benchList) {
+    const groupedBy = {};
+    benchList.forEach((obj) => {
+        if (!groupedBy[obj.benchmark]) {
+            groupedBy[obj.benchmark] = {};
+        }
+        groupedBy[obj.benchmark][obj.runMethod] = obj;
+    });
+    return groupedBy;
+}
+
+const compareKeys = ["# Instructions"];
+function buildEntry(run) {
+    const results = run.hyperfine.results[0];
+    return {
+        name: run.runMethod,
+        "# Instructions": run.total_dyn_inst,
+        min: tryRound(results.min),
+        max: tryRound(results.max),
+        mean: tryRound(results.mean),
+        median: tryRound(results.median),
+        stddev: tryRound(results.stddev),
+    }
+}
+
+function buildTableText(prevRun, run) {
+    const entry = buildEntry(run)
+    if (!prevRun) {
+        return entry;
+    }
+
+    const prevEntry = buildEntry(prevRun);
+    compareKeys.forEach((key) => {
+        const diff = Math.abs(entry[key] - prevEntry[key]);
+        if (diff === 0) {
+            return;
+        }
+
+        const sign = entry[key] < prevEntry[key] ? "-" : "+"
+        entry[key] = `${entry[key]} (${sign}${diff})`
+    })
+    return entry;
+}
+
+async function loadBenchmarks(compareTo) {
+    // data should be in the format of Array<{runMethod: String, benchmark: String, total_dyn_inst: Int, hyperfine: Array<{results: **hyperfine `--json` results**}>}>
+    console.log(data);
```

Comment:
where is `data` bound? not seeing it

Reply:
`data` is bound to a global that sits in `profile.js` which is an autogenerated file. Here's an example: https://nightly.cs.washington.edu/reports/eggcc/1700941844%3Anightly%3Anightly-comparison%3Ae51adfee2f/data/profile.js

and is generated here: https://github.com/egraphs-good/eggcc/blob/bd8e1f8168e8ae0ed6687dbd58b7e67e597c6406/infra/nightly.sh#L58

Reply:
Ohhh right thanks

Reply:
It seems weird that we scrape the other one we compare to but we have a global for this one. Shouldn't we get the data the same way for both?

Reply:
Good question. I think historically we were just making it easier on ourselves so we didn't have to deal with CORS and call `fetch` but now I think it would probably be best if we unified the way we do it. I'll add that to the PR, it isn't a lot of work